Assets/HoloToolkit/Input/Scripts/Utilities/CameraMotionInfo.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using HoloToolkit.Unity;
+
+namespace HoloToolkit.Unity.InputModule
+{
+    public class CameraMotionInfo : Singleton<CameraMotionInfo>
```

Comment:
Its currently in Input/Utilities, is there a better place for it?

Reply:
I'm also concerned we're duplicating raycast logic.

Reply:
@StephenHodgson yeah, agreed. the main thing here is the motion math. I'd be perfectly happy to axe *RayCastFromHead* and move *SphereCastFromHead* to *GazeManager*.

Reply:
So currently GazeManager is a mess. I saw some new checkin by @Railboy applying Ray[0] ray step changes and this has made that class quite unreadable. 

This logic could live in FocusManager perhaps?

Reply:
I currently have commits where *SphereCastFromHead* is folded into *GazeManager*. IMO it should probably belong there even if its currently a mess. *SphereCastFromHead* implements *RayStep* etc. I've gone ahead and cut *RayCastFromHead* since its basically just a wrapper for *GazeManager.Instance.HitInfo*.